### PR TITLE
Feature/Media Skip Indicators

### LINKED ISSUE #122: [SDL 0198] - Media Skip Indicators
Proposal: [Media Skip Indicators](https://github.com/smartdevicelink/sdl_evolution/blob/master/proposals/0198-media-skip-indicators.md)

> This feature is similar to [SDL-0109 Audio Streaming Indicator](https://github.com/smartdevicelink/sdl_evolution/blob/master/proposals/0109-set-audio-streaming-indicator.md) except that it affects the next and previous buttons. These buttons should have the ability to show time skip buttons that are commonly used by podcast & audiobook media. 

Review: https://github.com/smartdevicelink/sdl_evolution/issues/596

Steering Committee Decision:

> The Steering Committee voted to accept this proposal with the following revisions:
- Incorporate the feedback agreed to [in this comment](https://github.com/smartdevicelink/sdl_evolution/issues/596#issuecomment-418749992) by the author.
 - Ensure that conditions of `SeekIndicatorType` are documented, as requested [in this comment](https://github.com/smartdevicelink/sdl_evolution/issues/596#issuecomment-418755274). 

> The proposal .md file was updated to reflect these revisions on 9/20/18.

### PR DESCRIPTION
**Pull request by ShobhitAd**
Implements/Fixes #122

This PR is **ready** for review.

### Summary
- Added the new params from the HMI API
- Added new icons for indicator type TIME
- Implemented seek indicator icon switching

### CLA
- [x] I have signed [the CLA](https://docs.google.com/forms/d/e/1FAIpQLSdsgJY33VByaX482zHzi-xUm49JNnmuJOyAM6uegPQ2LXYVfA/viewform)

### CODE REVIEW

File: app/model/sdl/MediaModel.js [hunk 1]
```diff
@@ -114,7 +115,8 @@ SDL.SDLMediaModel = SDL.ABSAppModel.extend({
   isPlaying: false,
   isTemplate:false,
   mode:'',
-  
+  seekStreamingIndicator: {type: 'TRACK', seekTime: null},
```
[the remark this answers is not in the document]
Comment (ShobhitAd) [reply app/model/sdl/MediaModel.js]: Added separate properties for the for the forward and back seek indicators in https://github.com/smartdevicelink/sdl_hmi/pull/475/commits/29ad97868f95e81610e1e1a731e33611dea0bd1c

File: app/view/media/sdl/controllsView.js [hunk 2]
```diff
@@ -82,7 +82,21 @@ SDL.SDLMediaControlls = Em.ContainerView.create(
               'bc-item-big',
               'prevcd'
             ],
-            icon: 'images/media/ico_prew.png',
+            onIconChange: function() {
+              let icon = "images/media/ico_prew.png"; 
+              for (var i = 0; i < SDL.SDLModel.data.registeredApps.length; i++) {
+                appID = SDL.SDLModel.data.registeredApps[i].appID;
+                if (appID == SDL.SDLMediaController.currentAppId) {
```
Comment (iCollin) [inline app/view/media/sdl/controllsView.js:89]:
```suggestion
                if (SDL.SDLMediaController.currentAppId
                  == SDL.SDLModel.data.registeredApps[i].appID) { 
```

only read once, i think we should just compare it here

File: app/view/media/sdl/controllsView.js [hunk 3]
```diff
@@ -129,7 +143,21 @@ SDL.SDLMediaControlls = Em.ContainerView.create(
               'nextcd'
             ],
             classNameBindings: 'SDL.SDLController.model.SEEKRIGHT::unsubscribed',
-            icon: 'images/media/ico_next.png',
+            onIconChange: function() {
+              let icon = "images/media/ico_next.png"; 
+              for (var i = 0; i < SDL.SDLModel.data.registeredApps.length; i++) {
+                appID = SDL.SDLModel.data.registeredApps[i].appID;
+                if (appID == SDL.SDLMediaController.currentAppId) {
```
Comment (iCollin) [inline app/view/media/sdl/controllsView.js:150]:
```suggestion
                if (SDL.SDLMediaController.currentAppId
                  == SDL.SDLModel.data.registeredApps[i].appID) { 
```

**Comment by iCollin** [comment]
All is working great, but the SEEKTIME icon is slightly above vertical center in the button. The SEEKTRACK icon is as tall as the container, could we apply a similar css to the SEEKTIME to get it centered?

**Comment by ShobhitAd** [comment]
> All is working great, but the SEEKTIME icon is slightly above vertical center in the button. The SEEKTRACK icon is as tall as the container, could we apply a similar css to the SEEKTIME to get it centered?

Good catch! Modified the icon dimensions in https://github.com/smartdevicelink/sdl_hmi/pull/475/commits/012d63b07f016940c0905b46467b6af7c212ee96. Left me know if that looks good.

**Review by iCollin — APPROVED** [review]

**Review by iCollin — APPROVED** [review]

**Pull request merged**